engine/install/ubuntu.md

```diff
@@ -135,30 +121,27 @@ from the repository.
     <div id="x86_64_repo" class="tab-pane fade in active" markdown="1">
 
     ```bash
-    $ sudo add-apt-repository \
-       "deb [arch=amd64] {{ download-url-base }} \
-       $(lsb_release -cs) \
-       stable"
+    $ echo \
+      "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] {{ download-url-base }} \
+      $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
```

Comment:
not a blocker; mostly curious; what's the advantage of doing `echo | sudo tee` instead of `sudo echo > /etc/apt...` ? Is it so that `lsb_release -cs` is not executed as `root` ?

Reply:
`sudo echo >` does not work, as only the echo command itself had root permissions, while the redirect/file write itself has not and hence would fail. So `echo | sudo tee` is a necessity to grant root permissions for the file write itself, not a security measure.

`lsb_release` is always executed as original user, regardless if `$(lsb_release -cs)` is part of the sudo'ed command or any other part of the pipe. Like variables, command substitutions are expanded before the resulting string is passed to the parent command. Hence `sudo parent "$(child)"` would execute `child` as original user and pass the resulting string as argument to `parent` and only that is executed as root.

But indeed it could be seen as security measure to grant root permissions only to the final file write, so that, if the other commands were maliciously manipulated, they cannot do more than affecting the string that is written to a defined file. For the `gpg` command above that would be an argument (where a `sudo` is missing, I see now 😃):
```
sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
```
vs
```
gpg --dearmor | sudo tee /usr/share/keyrings/docker-archive-keyring.gpg
```
But I mean `tee` could be manipulated as well, and considering the other benefits that `-o` has (not creating an empty file when it fails), I would keep it the way it is now.

Reply:
This seems to work:
```
# gpg --fingerprint --no-default-keyring --keyring /etc/apt/trusted.gpg.d/debian-archive-buster-stable.gpg
/etc/apt/trusted.gpg.d/debian-archive-buster-stable.gpg
-------------------------------------------------------
pub   rsa4096 2019-02-05 [SC] [expires: 2027-02-03]
      6D33 866E DD8F FA41 C014  3AED DCC9 EFBF 77E1 1517
uid           [ unknown] Debian Stable Release Key (10/buster) <debian-release@lists.debian.org>
```

Reply:
> sudo echo > does not work, as only the echo command itself had root permissions, while the redirect/file write itself has not and hence would fail.

oh! you're right 🤦 

> This seems to work:

Thanks! I made some notes / possible approaches for checking the fingerprint (I recall there were some of those options that worked on the more "recent" versions of Debian/Ubuntu, but were not yet available on the older versions.

I'll dig up those notes, and will open a pull request later today